src/Strategy.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {SafeERC20, IERC20} from "src/Common.sol";
+import {BaseVault} from "src/BaseVault.sol";
+
+contract Strategy is BaseVault {
+    /**
+     * @notice Initializes the Strategy Vault.
+     * @param admin The address of the admin.
+     * @param name The name of the vault.
+     * @param symbol The symbol of the vault.
+     * @param decimals_ The number of decimals for the vault token.
+     */
+    function initialize(address admin, string memory name, string memory symbol, uint8 decimals_)
+        external
+        initializer
+    {
+        __ERC20_init(name, symbol);
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+
+        VaultStorage storage vaultStorage = _getVaultStorage();
+        vaultStorage.paused = true;
+        vaultStorage.decimals = decimals_;
+    }
+
+    /**
+     * @notice Returns the maximum amount of assets that can be withdrawn by a given owner.
+     * @param owner The address of the owner.
+     * @return uint256 The maximum amount of assets.
+     * @dev override the maxWithdraw function for strategies
+     */
+    function maxWithdraw(address owner) public view override returns (uint256) {
+        if (paused()) {
+            return 0;
+        }
+
+        uint256 ownerShares = balanceOf(owner);
+        uint256 maxAssets = convertToAssets(ownerShares);
+
+        return maxAssets;
+    }
+
+    /**
+     * @notice Returns the maximum amount of shares that can be redeemed by a given owner.
+     * @param owner The address of the owner.
+     * @return uint256 The maximum amount of shares.
+     * @dev override the maxRedeem function for strategies
+     */
+    function maxRedeem(address owner) public view override returns (uint256) {
+        if (paused()) {
+            return 0;
+        }
+
+        return balanceOf(owner);
+    }
+
+    /**
+     * @notice Internal function to handle deposits.
+     * @param asset_ The address of the asset.
+     * @param caller The address of the caller.
+     * @param receiver The address of the receiver.
+     * @param assets The amount of assets to deposit.
+     * @param shares The amount of shares to mint.
+     * @param baseAssets The base asset convertion of shares.
+     * @dev This is an example:
+     *     The _deposit function for strategies needs an override
+     */
+    function _deposit(
```

Comment:
why do we override this as well and remove the asset check?

Reply:
this was an example Strategy.sol, i didn't go into audit, we can improve it